awscli/customizations/assumerole.py

```diff
@@ -36,8 +38,41 @@ def inject_assume_role_provider_cache(session, **kwargs):
                   "assume-role cred provider cache.  Not configuring "
                   "JSONFileCache for assume-role.")
         return
-    provider = cred_chain.get_provider('assume-role')
-    provider.cache = JSONFileCache()
+    cred_chain.get_provider('assume-role').cache = JSONFileCache()
+    saml_provider = cred_chain.get_provider('assume-role-with-saml')
+    saml_provider.cache = JSONFileCache()
+    saml_provider.role_selector = role_selector
+
+
+def role_selector(role_arn, roles):
+    if role_arn:
+        chosen_roles = [r for r in roles if r['RoleArn']==role_arn]
+        if not chosen_roles:
+            raise ValueError(
+                'Your specified role %s does not exist in IdP roles list: %s'
+                % (role_arn, [r['RoleArn'] for r in roles]))
+        role = chosen_roles[0]
+    else:
+        # So end user has not yet configured a role_arn.
+        # If IdP provides more than one role, ask the user to choose one,
+        # otherwise just proceed.
+        if len(roles) > 1:
```

Comment:
I don't think we should be prompting users to select a role here.  The CLI typically only prompts for things when required (i.e MFA tokens and soon the SAML passwords).  They should explicitly set which role they want to use in their config file.

I like the AWS powershell model for how they handle SAML.  Configuration is a separate concern.  If we want to update our `aws configure` command to prompt for SAML values, do an initial login and prompt you to pick which role to associate with the profile, that seems pretty reasonable.